app/controllers/answers_controller.rb

```diff
@@ -41,7 +41,7 @@ def delete
        current_user.role == 'admin' ||
        current_user.role == 'moderator'
       respond_to do |format|
-        if @answer.delete
+        if @answer.destroy
```

Comment:
destroy method triggers dependent: :destroy callback, hence, delete was changed to destroy here.